madanalysis/layout/plotflow.py

```diff
@@ -110,865 +123,1118 @@ def InitializeHistoFrequency(self,ihisto):
+                    logging.getLogger("MA5").debug("Nominal weight location = " + str(nominal_loc))
+                    current_histo = np.squeeze(full_histo[:, nominal_loc])
+
+                # Scale variation envelope
+                if weight_set["weights"].has_scale:
+                    upper_histo = np.copy(current_histo)
+                    lower_histo = np.copy(current_histo)
+                    for x in scale_vars:
+                        scale_histo = np.squeeze(full_histo[:, x.loc])
+                        for i, mybin in enumerate(current_histo):
+                            if upper_histo[i] < scale_histo[i]: upper_histo[i] = scale_histo[i]
+                            if lower_histo[i] > scale_histo[i]: lower_histo[i] = scale_histo[i]
+                    upper_scale = np.abs(upper_histo - current_histo)
+                    lower_scale = np.abs(lower_histo - current_histo)
+                    scale_unc = np.vstack([lower_scale, upper_scale])
+
+                # PDF variations
+                if len(weight_set["pdf_variations"]) != 0:
+                    pdf_unc = {};
+                    for pdf_set, pdf_weights in weight_set["pdf_variations"].items():
+                        pdfvar_loc   = pdf_weights.loc + nominal_loc
+                        pdfvar_histo = full_histo[:, pdf_weights.loc]
+
+                        ### Method to use for PDF uncertainties
+                        ### TODO: verify that this works for all standard sets
+                        method = "replicas" if (("NNPDF" in pdf_set and not "hessian" in pdf_set) or ("PDF4LHC" in pdf_set and "_mc_" in pdf_set)) else "hessian"
+                        logging.getLogger("MA5").debug(f"Using {method} pdf combination for {pdf_set} pdf set.")
+
+                        ### Replicas method
+                        if method == "replicas":
+                            mean_histo = np.mean(pdfvar_histo, axis=1)
+                            uncertainties = np.sqrt(np.sum(np.square(pdfvar_histo - mean_histo.reshape(-1, 1)),axis=1)/pdfvar_histo.shape[1])
+                        ### Hessian method
+                        else:
+                            uncertainties = np.sqrt(np.sum(np.square(pdfvar_histo - current_histo.reshape(-1, 1)),axis=1))
+                        pdf_unc[pdf_set] = np.vstack([uncertainties, uncertainties])
+
+            # TODO give options for both linear and quadrature combination
+            total_unc = None
+            # Two sets of uncertainties
+            if scale_unc is not None and pdf_unc is not None:
+                lower_unc = scale_unc[0]
+                upper_unc = scale_unc[1]
+                for pdf_set, pdf_error in pdf_unc.items():
+                    lower = np.sqrt(np.square(scale_unc[0]) + np.square(pdf_error[0]))
+                    upper = np.sqrt(np.square(scale_unc[1]) + np.square(pdf_error[1]))
```

Comment:
Are you combining them in quadrature here? They are at the absolute error stage here, not fractions. Shouldn't it be divided by nominal error first?

Reply:
This should not matter as we re-multiply by the nominal weight afterwards, shouldn't it?

Reply:
I'm going to get more coffee...

Reply:
At this stage you may just take pure coffee, without water ;)